src/sentry/uptime/endpoints/validators.py

```diff
@@ -463,44 +463,90 @@ def create_source(self, validated_data: dict[str, Any]) -> UptimeSubscription:
         return uptime_subscription
 
 
-class UptimeDomainCheckFailureValidator(BaseDetectorTypeValidator):
-    enforce_single_datasource = True
-    data_sources = serializers.ListField(child=UptimeMonitorDataSourceValidator(), required=False)
+class UptimeDomainCheckFailureConfigValidator(CamelSnakeSerializer):
```

Comment:
hmm, is there a need to move away from the base validator? are there updates here that we should move over to that to improve all of the detector validators?